File: solutions/811.number-of-subarrays-with-bounded-maximum/number-of-subarrays-with-bounded-maximum.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int>& A, int L, int R) {
        int ans = 0;
        int left = -1;
        int right = -1;
        for (int i = 0; i < A.size(); ++i) {
          if (A[i] >= L) right = i;
          if (A[i] > R) left = i;      
          ans += (right - left);
        }
        return ans;
    }
};

File: solutions/811.number-of-subarrays-with-bounded-maximum/number-of-subarrays-with-bounded-maximum.java
class Solution {
    
    public int numSubarrayBoundedMax(int[] A, int L, int R) {
        return count(A, R) - count(A, L - 1);
    }
    
    private int count(int[] A, int N) {
        int res = 0, cur = 0;
        for(int num : A) 
            if(num <= N) res += ++cur;
            else cur = 0;
        return res;
    }
        
}